src/test/java/ch/uzh/ifi/fs22/sel/group15/cozycave/server/service/UserServiceTest.java

```diff
@@ -0,0 +1,176 @@
+import ch.uzh.ifi.fs22.sel.group15.cozycave.server.entity.Location;
+import ch.uzh.ifi.fs22.sel.group15.cozycave.server.entity.Picture;
+import ch.uzh.ifi.fs22.sel.group15.cozycave.server.entity.user.*;
+import ch.uzh.ifi.fs22.sel.group15.cozycave.server.repository.UserRepository;
+import ch.uzh.ifi.fs22.sel.group15.cozycave.server.service.UserService;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.Date;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.BDDMockito.given;
+
+class UserServiceTest {
+
+
+    @Mock
+    private UserRepository userRepository;
+
+    @InjectMocks
+    private UserService userService;
+
+    private User testUser;
+    private AuthenticationData testAuthenticationData;
+    private UserDetails testUserDetails;
+    private Location testLocation;
+
+    // create test user before each setup
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.openMocks(this);
+
+        Date dNow = new Date();
+        UUID userId = UUID.randomUUID();
+        UUID locationId = UUID.randomUUID();
+        testLocation = new Location(locationId, "Test", "TestAddress", "Mainstreet", "1", 8000, "Zurich", "Switzerland");
+        testAuthenticationData = new AuthenticationData(userId, "testUser@uzh.ch", "Test1234!?", "Test1234!?", "Test1234!?");
+        testUserDetails = new UserDetails(userId, "Test", "User", Gender.OTHER, dNow, testLocation, "Simple Bioagraphy of the TestUser");
+        testUser = new User(userId, dNow, testAuthenticationData, Role.STUDENT, testUserDetails);
+
+        Mockito.when(userRepository.save(Mockito.any())).thenReturn(testUser);
+    }
+
+    @Test
+    public void createUserSuccess() {
+        User createdUser = userService.createUser(testUser);
+        Mockito.verify(userRepository, Mockito.times(1)).save(Mockito.any());
+        // compare emails of user
+        assertEquals(testUser.getAuthenticationData().getEmail(), createdUser.getAuthenticationData().getEmail());
+    }
+
+    @Test
+    public void createUserEmpty() {
+
+    }
+
+    @Test
+    public void createUserDuplicateEmailException() {
+        userService.createUser(testUser);
+
+        // when -> setup additional mocks for UserRepository
+        Mockito.when(userRepository.findByAuthenticationData_Email(Mockito.any())).thenReturn(Optional.ofNullable(testUser));
+        Mockito.when(userRepository.findByAuthenticationData_Email(Mockito.any())).thenReturn(null);
```

Comment:
second call overwrites first, line 77 probably should be something like: 

`Mockito.when(userRepository.findByAuthenticationData_Email("mail-of-testuser@test.com")).thenReturn(Optional.ofNullable(testUser));`